eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -115,14 +148,176 @@ private VariantType getVariantType(String reference, String alternate) {
         return variantType;
     }
 
+    private void merge(Long providedAccession, String hash, Long accessionInDatabase) {
+        Priority prioritised = prioritise(providedAccession, accessionInDatabase);
+
+        Query queryClustered = query(where(ACCESSION_KEY).is(prioritised.accessionToBeMerged));
+        List<? extends ClusteredVariantEntity> cvToMerge =
+                mongoTemplate.find(queryClustered, getClusteredVariantCollection(prioritised.accessionToBeMerged));
+
+        List<? extends ClusteredVariantEntity> cvToKeep =
+                mongoTemplate.find(query(where(ACCESSION_KEY).is(prioritised.accessionToKeep)),
+                                   getClusteredVariantCollection(prioritised.accessionToKeep));
+
+        if (isMultimap(cvToMerge) || isMultimap(cvToKeep)) {
+            // multimap! don't merge. see isMultimap() below for more details
+            return;
+        }
+
+        assignedAccessions.put(hash, prioritised.accessionToKeep);
+
+        // write operations for clustered variant being merged
+        List<ClusteredVariantOperationEntity> operations =
+                cvToMerge.stream()
+                         .map(c -> buildClusteredOperation(c, prioritised.accessionToKeep))
+                         .collect(Collectors.toList());
+        mongoTemplate.insert(operations, getClusteredOperationCollection(prioritised.accessionToBeMerged));
+
+        mongoTemplate.updateMulti(queryClustered, update(ACCESSION_KEY, prioritised.accessionToKeep),
+                                  getClusteredVariantCollection(prioritised.accessionToBeMerged));
+
+        // Update submitted variants linked to the clustered variant we just merged.
+        // This has to happen for both EVA and dbsnp SS because previous cross merges might have happened.
+        updateSubmittedVariants(prioritised, SubmittedVariantEntity.class, SubmittedVariantOperationEntity.class);
+        updateSubmittedVariants(prioritised, DbsnpSubmittedVariantEntity.class,
+                                DbsnpSubmittedVariantOperationEntity.class);
+    }
+
+    private Class<? extends ClusteredVariantEntity> getClusteredVariantCollection(Long accession) {
```

Comment:
May be this is already part of polishing but I feel that this and the following function should belong in a separate class since they represent a business rule

Reply:
I don't really understand why business rules should be treated differently than the rest of the code, but I see the value of extracting it to it's own class, maybe even in eva-accession-core. I'll add a note in the ticket for polishing.